l5kit/l5kit/sampling/agent_sampling.py

```diff
@@ -87,20 +70,29 @@ def generate_agent_sample(
     history_frames = frames[history_slice].copy()  # copy() required if the object is a np.ndarray
     future_frames = frames[future_slice].copy()
 
-    min_agents_index = history_frames[-1]["agent_index_interval"][0]
-    if len(future_frames) > 0:
-        max_agents_index = future_frames[-1]["agent_index_interval"][1]
-    else:  # future_frames can be empty at scene edges
-        max_agents_index = history_frames[0]["agent_index_interval"][1]
+    sorted_frames = np.concatenate((history_frames[::-1], future_frames))  # from past to future
 
+    # get agents (past and future)
+    min_agents_index = sorted_frames[0]["agent_index_interval"][0]
+    max_agents_index = sorted_frames[-1]["agent_index_interval"][1]
     agents = agents[min_agents_index:max_agents_index].copy()  # this is the minimum slice of agents we need
-
     history_frames["agent_index_interval"] -= min_agents_index  # sync interval with the agents array
     future_frames["agent_index_interval"] -= min_agents_index  # sync interval with the agents array
-
     history_agents = filter_agents_by_frames(history_frames, agents)
     future_agents = filter_agents_by_frames(future_frames, agents)
 
+    try:
+        min_tl_index = history_frames[-1]["traffic_light_faces_index_interval"][0]  # -1 is the farthest in the past
+        max_tl_index = history_frames[0]["traffic_light_faces_index_interval"][1]
```

Comment:
Might be good to again define these magic strings in MapAPI to avoid possible undetectable copy-paste issues.